scripts/lib/CIME/XML/env_batch.py

```diff
@@ -270,6 +269,10 @@ def get_submit_args(self, case, job):
                     if flag == "-n" and rval<= 0:
                         rval = 1
 
+                    if flag == "-q" and rval == "batch" and case.get_value("MACH") == "blues":
```

Comment:
@jedwards4b I couldn't figure out how to do that in XML. It's a pretty unusual case: the -q <queue> option needs to be provided unless the queue is "batch". I don't think we can express that in XML.